instrumentation/kafka-clients/kafka-clients-2.6/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/KafkaTracingBuilder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.kafkaclients;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.messaging.MessageOperation;
+import io.opentelemetry.instrumentation.kafka.KafkaInstrumenterBuilder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+public class KafkaTracingBuilder {
+  private static final String INSTRUMENTATION_NAME = "io.opentelemetry.kafka-clients-0.11";
+
+  private final OpenTelemetry openTelemetry;
+  private final List<AttributesExtractor<ProducerRecord<?, ?>, Void>> producerAttributesExtractors =
+      new ArrayList<>();
+  private final List<AttributesExtractor<ConsumerRecord<?, ?>, Void>> consumerAttributesExtractors =
+      new ArrayList<>();
+
+  KafkaTracingBuilder(OpenTelemetry openTelemetry) {
+    this.openTelemetry = Objects.requireNonNull(openTelemetry);
+  }
+
+  public void addProducerAttributesExtractors(
+      AttributesExtractor<ProducerRecord<?, ?>, Void> extractor) {
+    producerAttributesExtractors.add(extractor);
+  }
+
+  public void addConsumerAttributesExtractors(
+      AttributesExtractor<ConsumerRecord<?, ?>, Void> extractor) {
+    consumerAttributesExtractors.add(extractor);
+  }
```

Comment:
These methods should probably return `this`, we usually use fluent-style builders in our codebase:
```suggestion
  public KafkaTracingBuilder addProducerAttributesExtractors(
      AttributesExtractor<ProducerRecord<?, ?>, Void> extractor) {
    producerAttributesExtractors.add(extractor);
    return this;
  }

  public KafkaTracingBuilder addConsumerAttributesExtractors(
      AttributesExtractor<ConsumerRecord<?, ?>, Void> extractor) {
    consumerAttributesExtractors.add(extractor);
    return this;
  }
```